CommonTools/Utils/interface/KinematicTables.h

```diff
@@ -22,6 +22,16 @@ namespace edm {
                                        col::Phi::filler([](Object const& x) { return x.phi(); }))};
     }
 
+    using PtEtaPhiTable = edm::soa::Table<col::Pt, col::Eta, col::Phi>;
+
+    template <class Object>
+    PtEtaPhiTable makePtEtaPhiTable(std::vector<Object> const& objects) {
+      return {objects,
+              edm::soa::column_fillers(col::Pt::filler([](Object const& x) { return x.pt(); }),
+                                       col::Eta::filler([](Object const& x) { return x.eta(); }),
+                                       col::Phi::filler([](Object const& x) { return x.phi(); }))};
+    }
+
     template <class Object>
     auto makeEtaPhiTableLazy(std::vector<Object> const& objects) {
```

Comment:
this is kind of left dangling; it would be better to have a more continuous implementation: etaPhi first, then ptEtaPhi.

Reply:
pinging @guitargeek as the author of `KinematicTables` - any suggestions as to how I could improve this would be much appreciated!

Reply:
Hi Joosep, I think Slavas comment refers just to the order of the statements in this file. It's a bit confusing that there is there first the `makeEtaPhiTable` function, then the `makePtEtaPhiTable`, and then it goes back to only EtaPhi with `makeEtaPhiTableLazy`.

Other than this possible reordering, I would not bother changing anything else in this file in this PR, in particular not anything related to the column fillers. This is under development already in https://github.com/cms-sw/cmssw/pull/30956. We can just rebase whichever PR gets merged second (it will probably be mine). Is that okay for you?

Reply:
> Hi Joosep, I think Slavas comment refers just to the order of the statements in this file, right? It's a bit confusing that there is there first the `makeEtaPhiTable` function, then the `makePtEtaPhiTable`, and then it goes back to only EtaPhi with `makeEtaPhiTableLazy`.

right, that's what I meant

Reply:
Oh, got it, thanks. I thought the idea is to reduce the duplication a bit (don't have a good idea for it immediately).

Reply:
looking at https://github.com/cms-sw/cmssw/blob/master/FWCore/SOA/interface/Table.h

```using PtEtaPhiTable = AddColumns_t<EtaPhiTable, col::Pt>;```

appears to be the direction to go with reusing the available EtaPhi.
I'm not sure about the related methods if there is a transparent way to extend

Reply:
Very soon these functions won't be necessary anyway. When de default fillers are implemented with #30956, the only justification for their existence is to support the lazy construction via `LazyResult`. While this `LazyResult` wrapper is nice, I have now realized that it can't be conveniently used to construct objects without the `makeEtaPhiTableLazy` wrapper function in this file.

We should maybe consider implementing a `LazyConstructed` wrapper to simplify this (maybe in my PR once I solved the other problems). Then, in the producers one would use the table as:
```C++
auto etaPhiTable = LazyConstructed<edm::soa::EtaPhiTable>(tracks);
```
This `KinematicTables.h` file would then only have the `using` statements, where the tables build on top of each other with `AddColumns_t` as Slava said.